src/cudamatrix/cu-array-inl.h

```diff
@@ -119,17 +119,34 @@ void CuArray<T>::CopyToVec(std::vector<T> *dst) const {
 #if HAVE_CUDA == 1
   if (CuDevice::Instantiate().Enabled()) { 
     Timer tim;
-    CU_SAFE_CALL(cudaMemcpy(&dst->front(), Data(), dim_*sizeof(T), cudaMemcpyDeviceToHost));
+    CU_SAFE_CALL(cudaMemcpy(&dst->front(), Data(), dim_ * sizeof(T), cudaMemcpyDeviceToHost));
     CuDevice::Instantiate().AccuProfile("CuArray::CopyToVecD2H", tim.Elapsed());
   } else
 #endif
   {
-    memcpy(&dst->front(), data_, dim_*sizeof(T));
+    memcpy(&dst->front(), data_, dim_ * sizeof(T));
   }
 }
 
 
 template<typename T>
+void CuArray<T>::CopyToHost(T *dst) const {
+  if (dst == NULL) return;
```

Comment:
You should be asserting that dst != NULL instead of just returning if it's NULL -- but do this after the statement
if (dim_ == 0) return;